imap/annotate.c

```diff
@@ -1200,22 +1202,22 @@ static int annotatemore_findall_full(const char *pattern, /* internal */
     frock.since_modseq = since_modseq;
     frock.flags = flags;
 
-    r = _annotate_getdb(mailbox ? mailbox_uniqueid(mailbox) : NULL, mailbox, uid, 0, &frock.d);
+    if (mailbox || mboxname) {
```

Comment:
Something like this here

```suggestion
    if (mailbox || (mboxname && *mboxname)) {
```

sorts out the first part of the "mboxname might be `""`" problem (but is incomplete, more below)

Reply:
Yep, fixed this up so the other variables are right too